test/integration/install/inject/inject_test.go

```diff
@@ -481,29 +482,79 @@ func TestInjectAutoPod(t *testing.T) {
 			testutil.Fatalf(t, "expected pod in namespace %s to have %s opaque ports, but it had %s", ns, manualOpaquePorts, annotation)
 		}
 
+		initContainerPods, err := TestHelper.GetPods(ctx, ns, map[string]string{"app": initContainerPodName})
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "failed to get pods", "failed to get pods for namespace %s: %s", ns, err)
+		}
+		if len(opaquePods) != 1 {
```

Comment:
```suggestion
		if len(initContainerPods) != 1 {
```

Reply:
Good catch!

Fixed in https://github.com/linkerd/linkerd2/pull/8034/commits/d34e55e5ea31f1d69f92017f2841fe6b27b63c48